apps/src/code-studio/components/progress/ScriptTeacherPanel.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import TeacherPanel from '../TeacherPanel';
+import ToggleGroup from '@cdo/apps/templates/ToggleGroup';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import { ViewType, setViewType, selectSection } from '../../stageLockRedux';
+
+const styles = {
+  viewAs: {
+    fontSize: 16,
+    margin: 10
+  },
+  toggleGroup: {
+    margin: 10
+  },
+  select: {
+    margin: 10,
+    width: 180
+  },
+  text: {
+    margin: 10
+  },
+  exclamation: {
+    color: 'red'
+  },
+  dontForget: {
+    display: 'inline',
+    marginLeft: 10,
+    fontSize: 16,
+    fontFamily: '"Gotham 7r", sans-serif'
+  }
+};
+
+const ScriptTeacherPanel = React.createClass({
+  propTypes: {
+    viewAs: React.PropTypes.oneOf(Object.values(ViewType)).isRequired,
+    sections: React.PropTypes.objectOf(
+      React.PropTypes.shape({
+        section_name: React.PropTypes.string.isRequired
+      })
+    ).isRequired,
+    selectedSection: React.PropTypes.string,
+    sectionsLoaded: React.PropTypes.bool.isRequired,
+    unlockedStageNames: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+    setViewType: React.PropTypes.func.isRequired,
+    selectSection: React.PropTypes.func.isRequired,
+  },
+
+  handleSelectChange(event) {
+    this.props.selectSection(event.target.value);
+  },
+
+  render() {
+    const {
+      viewAs,
+      sections,
+      selectedSection,
+      sectionsLoaded,
+      setViewType,
+      unlockedStageNames
+    } = this.props;
+    const hasSections = Object.keys(sections).length > 0;
+    // TODO - i18n
+    // TODO - don't forget section is conditional on having unlocked stages
+    return (
+      <TeacherPanel>
+        <h3>Teacher Panel</h3>
+        <div className="content">
+          <div className="non-scrollable-wrapper">
+            <div style={styles.viewAs}>
+              View page as:
+            </div>
+            <div style={styles.toggleGroup}>
+              <ToggleGroup
+                selected={viewAs}
+                onChange={setViewType}
+              >
+                <button value={ViewType.Student}>Student</button>
+                <button value={ViewType.Teacher}>Teacher</button>
+              </ToggleGroup>
+            </div>
+          </div>
```

Comment:
This label/control could be easily extracted to a `ViewAsToggle` component.  Even inline in this file, it'd probably improve the readability of the render method.